code/modules/explorer_drone/adventure.dm

```diff
@@ -0,0 +1,422 @@
+// json field definitions bit verbose but i've had it with the typos
+#define ADVENTURE_NAME_FIELD "adventure_name"
+#define ADVENTURE_STARTING_NODE_FIELD "starting_node"
+#define ADVENTURE_REQUIRED_SITE_TRAITS_FIELD "required_site_traits"
+#define ADVENTURE_SCAN_BAND_MODS_FIELD "scan_band_mods"
+#define ADVENTURE_LOOT_FIELD "loot_categories"
+#define ADVENTURE_STARTING_QUALITIES_FIELD "starting_qualities"
+#define ADVENTURE_DEEP_SCAN_DESCRIPTION "deep_scan_description"
+#define ADVENTURE_NODES_FIELD "nodes"
+#define ADVENTURE_TRIGGERS_FIELD "triggers"
+
+#define NODE_NAME_FIELD "name"
+#define NODE_DESCRIPTION_FIELD "description"
+#define NODE_IMAGE_FIELD "image"
+#define NODE_RAW_IMAGE_FIELD "raw_image"
+#define NODE_CHOICES_FIELD "choices"
+#define NODE_ON_ENTER_EFFECTS_FIELD "on_enter_effects"
+#define NODE_ON_EXIT_EFFECTS_FIELD "on_exit_effects"
+
+#define CHOICE_KEY_FIELD "key"
+#define CHOICE_NAME_FIELD "name"
+#define CHOICE_ON_SELECTION_EFFECT_FIELD "on_selection_effects"
+#define CHOICE_REQUIREMENTS_FIELD "requirements"
+#define CHOICE_EXIT_NODE_FIELD "exit_node"
+#define CHOICE_DELAY_FIELD "delay"
+#define CHOICE_DELAY_MESSAGE_FIELD "delay_message"
+
+#define EFFECT_TYPE_FIELD "effect_type"
+#define EFFECT_QUALITY_FIELD "quality"
+#define EFFECT_VALUE_FIELD "value"
+#define EFFECT_VALUE_VALUE_TYPE_FIELD "value_type"
+#define TRIGGER_NAME_FIELD "name"
+#define TRIGGER_REQUIREMENTS_FIELD "requirements"
+#define TRIGGER_ON_TRIGGER_EFFECTS_FIELD "on_trigger_effects"
+#define TRIGGER_TARGET_NODE_FIELD "target_node"
+
+#define REQ_GROUP_REQUIREMENTS_FIELD "requirements"
+#define REQ_GROUP_GROUP_TYPE_FIELD "group_type"
+
+#define REQ_QUALITY_FIELD "quality"
+#define REQ_VALUE_FIELD "value"
+#define REQ_OPERATOR_FIELD "operator"
+
+GLOBAL_LIST_EMPTY(explorer_drone_adventures)
+
+
+/proc/load_adventures()
+	. = list()
+	for(var/filename in flist(ADVENTURE_DIR))
+		var/datum/adventure/A = try_loading_adventure(filename)
+		if(A)
+			. += A
+
+/proc/try_loading_adventure(filename)
+	var/list/json_data = json_load(ADVENTURE_DIR+filename)
+	if(!islist(json_data))
+		CRASH("Invalid JSON in adventure file [replacetext(filename,ADVENTURE_DIR,"")]")
+	//Basic validation of required fields, don't even bother loading if they are missing.
+	var/static/list/required_fields = list(ADVENTURE_NAME_FIELD,ADVENTURE_STARTING_NODE_FIELD,ADVENTURE_NODES_FIELD)
+	for(var/field in required_fields)
```

Comment:
Not for now, but maybe in the future we can make this a lint instead, for zero extra runtime cost. Oh well, not important since it's only done once anyway.